anitya/lib/backends/__init__.py

```diff
@@ -329,15 +329,16 @@ def get_versions_by_regex_for_text(text, url, regex, project):
             "%s: invalid regular expression" % project.name)
 
     for index, version in enumerate(upstream_versions):
-        # Strip the version_prefix early
-        version = version.lstrip(project.version_prefix)
         upstream_versions[index] = version
 
         # If the version retrieved is a tuple, re-constitute it
         if type(version) == tuple:
             version = ".".join([v for v in version if not v == ""])
             upstream_versions[index] = version
 
+        # Strip the version_prefix early
+        version = version.lstrip(project.version_prefix)
```

Comment:
hm, indeed, not sure why I didn't move that line when I moved the first two.

Thanks for catching this!